benchmarks-api/src/main/java/uk/co/real_logic/benchmarks/remote/InMemoryMessageTransceiver.java

```diff
@@ -23,59 +23,15 @@
 import static sun.misc.Unsafe.ARRAY_LONG_BASE_OFFSET;
 import static sun.misc.Unsafe.ARRAY_LONG_INDEX_SCALE;
 
-@SuppressWarnings("unused")
-abstract class InMemoryMessageTransceiver1 extends MessageTransceiver
+public final class InMemoryMessageTransceiver extends MessageTransceiver
 {
     static final int SIZE = 4096;
     private static final int MASK = SIZE - 1;
     private static final int SHIFT = 31 - numberOfLeadingZeros(ARRAY_LONG_INDEX_SCALE);
-    static final int PADDING = CACHE_LINE_LENGTH / SIZE_OF_LONG - 1;
-    final long[] messages = new long[SIZE];
+    private static final int PADDING = CACHE_LINE_LENGTH / SIZE_OF_LONG - 1;
+    private final long[] messages = new long[SIZE];
 
-    // Padding
-    boolean p000, p001, p002, p003, p004, p005, p006, p007, p008, p009, p010, p011, p012, p013, p014, p015;
```

Comment:
One cache line of padding is sufficient. Later processor are wiser about prefetching.

Reply:
I've remove padding completely, because the send and receive is done on one thread.